cognite/neat/rules/exporter/_rules2dms.py

```diff
@@ -546,171 +566,312 @@ def to_cdf(
+            )
         if "data model" in components_to_create or "all" in components_to_create:
-            self.create_data_model(client, existing_data_model)
+            logs["data model"], errors["data model"] = self.create_data_model(
+                client, existing_data_model, existing_component_handling == "update"
+            )
 
-    def create_space(self, client: CogniteClient, existing_spaces: set):
-        non_existing_spaces = self.spaces - existing_spaces
-        logging.info(f"Creating space {non_existing_spaces} !!!")
-        _ = client.data_modeling.spaces.apply([SpaceApply(space=space) for space in non_existing_spaces])
+        if return_report:
+            return logs, errors
 
-    def create_containers(self, client: CogniteClient, existing_containers: set):
-        non_existing_containers = set(self.containers.keys()) - existing_containers
-        logging.info(f"Creating container {non_existing_containers}")
-        _ = client.data_modeling.containers.apply([self.containers[id_] for id_ in non_existing_containers])
+    def create_space(
+        self,
+        client: CogniteClient,
+        existing_spaces: set,
+        update: bool = False,
+        multi_space_components_create: bool = False,
+    ) -> tuple[list, list]:
+        logs, errors = [], []
+
+        spaces_to_create = (
+            self.spaces - existing_spaces
+            if multi_space_components_create
+            else ((self.spaces - existing_spaces) & {self.space})
+        )
+
+        spaces_to_update = existing_spaces if multi_space_components_create else (existing_spaces & {self.space})
+
+        if spaces_to_create:
+            try:
+                _ = client.data_modeling.spaces.apply([SpaceApply(space=space) for space in spaces_to_create])
+                logs.append(f"Created space {spaces_to_create}")
+            except CogniteAPIError as e:
+                errors.append(f"Failed to create space {spaces_to_create}! Reason: {e.message}")
+
+        if update and spaces_to_update:
+            try:
+                _ = client.data_modeling.spaces.apply([SpaceApply(space=space) for space in spaces_to_update])
+                logs.append(f"Updated space {spaces_to_update}")
+            except CogniteAPIError as e:
+                errors.append(f"Failed to update spaces {spaces_to_update}! Reason: {e.message}")
+
+        return logs, errors
+
+    def create_containers(
+        self,
+        client: CogniteClient,
+        existing_containers: set,
+        update: bool = False,
+        multi_space_components_create: bool = False,
+    ) -> tuple[list, list]:
+        logs, errors = [], []
+
+        containers_to_create = (
+            set(self.containers.keys()) - existing_containers
+            if multi_space_components_create
+            else {k for k in (set(self.containers.keys()) - existing_containers) if k.split(":")[0] == self.space}
```

Comment:
Note sure if this is the PR, but code like this is fairly risky `k.split(":")[0] == self.space`. Suggest that we use `ContainerId` instead

Reply:
Let's hit this one in another PR, I was not extremely happy with this either, but will solves issues now/unblocks clients